plugins/services/src/js/components/forms/GeneralServiceFormSection.js

```diff
@@ -227,6 +233,15 @@ class GeneralServiceFormSection extends Component {
         <p>Constraints control where apps run to allow optimization for either fault tolerance or locality.</p>
         {this.getPlacementConstraintsFields(data.constraints)}
         <FormRow>
+          <FormGroup
+            className="column-12"
+            showError={constraintsErrors != null && !Array.isArray(constraintsErrors)}>
+            <FieldError>
+              {constraintsErrors}
+            </FieldError>
+          </FormGroup>
+        </FormRow>
```

Comment:
I removed the extra row. This what it looks like with an error message, which is exactly how I imagined it to be:
![](https://cl.ly/1z28203F0O0Q/Image%202017-02-24%20at%2019.08.49.png)

It also looks right without an error message showing.